source/extensions/filters/common/ratelimit_config/ratelimit_config.cc

```diff
@@ -111,6 +140,41 @@ void RateLimitPolicy::populateDescriptors(const Http::RequestHeaderMap& headers,
       descriptor.entries_.emplace_back(std::move(entry));
     }
   }
+
+  // Populate hits_addend if set.
+  if (hits_addend_provider_ != nullptr) {
+    const ProtobufWkt::Value hits_addend_value =
+        hits_addend_provider_->formatValueWithContext({&headers}, stream_info);
+
+    double hits_addend = 0;
+    bool success = true;
+
+    if (hits_addend_value.has_number_value()) {
+      hits_addend = hits_addend_value.number_value();
+    } else if (hits_addend_value.has_string_value()) {
+      // Attempt to parse the string as a double.
+      success = absl::SimpleAtod(hits_addend_value.string_value(), &hits_addend);
+    } else {
+      // Only number and string values are allowed.
+      success = false;
+    }
+
+    // Check value range.
+    if (hits_addend < 0 || hits_addend > MAX_HITS_ADDEND) {
+      success = false;
+    }
+
+    if (success) {
+      descriptor.hits_addend_ = static_cast<uint64_t>(hits_addend);
+    } else {
+      ENVOY_LOG(debug, "Invalid hits_addend: {}", hits_addend_value.DebugString());
```

Comment:
I think this is going to be very confusing to debug and I would go back to WARN but use one of the power of 2 or timed variants for that, but up to you.

Reply:
> I think this is going to be very confusing to debug and I would go back to WARN but use one of the power of 2 or timed variants for that, but up to you.

Will change this in next PR. There is a series of works.